File: app/src/main/java/com/stevenfrew/beatprompter/FolderFilter.java
package com.stevenfrew.beatprompter;

import java.util.ArrayList;

class FolderFilter extends SongFilter
{
    FolderFilter(String folderName,ArrayList<SongFile> songs)
    {
        super(folderName,songs,true);
    }

    public boolean equals(Object o) {
        return o instanceof FolderFilter && mName.equals(((FolderFilter) o).mName);
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/ui/filter/FolderFilter.kt
package com.stevenfrew.beatprompter.ui.filter

import com.stevenfrew.beatprompter.cache.SongFile

class FolderFilter(
	folderName: String,
	songs: List<SongFile>
) : SongFilter(folderName, songs, true) {
	override fun equals(other: Any?): Boolean = other is FolderFilter && name == other.name
	override fun hashCode(): Int = javaClass.hashCode()
}